File: Xamarin/Xamarin.SingleView/ViewController.cs
using System;
using UIKit;
using Binding;

namespace Xamarin.SingleView
{
    public partial class ViewController : UIViewController
    {
        private SwiftFrameworkProxy _proxy;

        public ViewController(IntPtr handle)
            : base(handle)
        {
        }

        public override void ViewDidLoad()
        {
            base.ViewDidLoad();

            btnLogin.TouchUpInside += btnLogin_Tap;
            _proxy = new SwiftFrameworkProxy();
            var result = _proxy.InitForApiKey("APIKey");
            System.Diagnostics.Debug.WriteLine(result);
            lblMessage.Text = result;
        }

        private void btnLogin_Tap(object sender, EventArgs e)
        {
            _proxy.LoginWithProvider(GigyaSocialProvidersProxy.Google, this);
        }
    }
}

File: Swift/Swift.SingleView/Swift.SingleView/ViewController.swift
//
//  ViewController.swift
//  Swift.SingleView
//
//  Created by Alexey Strakh on 10/4/19.
//  Copyright © 2019 Xamarin. All rights reserved.
//

import UIKit
import Gigya
import SwiftFrameworkProxy

class ViewController: UIViewController {

    @IBOutlet weak var lblMessage: UILabel!
    @IBOutlet weak var btnLoginWithGigya: UIButton!
    @IBOutlet weak var btnLoginWithProxy: UIButton!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        btnLoginWithGigya.addTarget(self, action: #selector(self.btnLoginWithGigya_Tap), for: .touchUpInside)
        btnLoginWithProxy.addTarget(self, action: #selector(self.btnLoginWithProxy_Tap), for: .touchUpInside)
      }

    @objc func btnLoginWithGigya_Tap(sender: UIButton!) {
        // init
        Gigya.sharedInstance().initFor(apiKey: "APIKey");
        let gigyaDomain = Gigya.sharedInstance().config.apiDomain
        let result = "Gigya initialized with domain: \(gigyaDomain)"
        lblMessage.text = result;
        
        // activate login
        Gigya.sharedInstance().login(
                   with: GigyaSocialProviders.google,
                   viewController: self,
                   completion: { _ in
                       // Gigya.login.completed
                   })
    }
    
    @objc func btnLoginWithProxy_Tap(sender: UIButton!) {
        // init
        let proxy = SwiftFrameworkProxy()
        let result = proxy.initFor(apiKey: "APIKey")
        
        // activate login
        lblMessage.text = result;
        proxy.login(
            provider: GigyaSocialProvidersProxy.google,
            viewController: self,
            completion: { (_, _) in
               // Gigya.login.completed
        });
      }
}

